sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -918,4 +1043,26 @@ public static void ResetFeatureSwitch()
                 "AZURE_EXPERIMENTAL_ENABLE_ACTIVITY_SOURCE");
         }
     }
+#else
+#pragma warning disable SA1507 // File can not contain multiple types
+    /// <summary>
+    /// Until Activity Source is no longer considered experimental.
+    /// </summary>
+    internal static class ActivityExtensions
+    {
+        private static bool SupportsActivitySourceSwitch;
+
+        public static bool SupportsActivitySource()
+        {
+            return SupportsActivitySourceSwitch;
+        }
+
+        public static void ResetFeatureSwitch()
```

Comment:
It's called by tests only, but in a few places (Event Hubs tests, Service Bus tests, Azure Core tests)

Reply:
It'd be pretty straightforward to move the body of this method to where it's being used, it's only called like 6 times, but I was hesitant because of consistency with the string values. I think it would be possible though to keep these string values as constants somewhere else. Maybe even just as part of the DiagnosticScope class.

Reply:
I think it's fine as is since it's called from multiple test projects.